src/analysis/normed/ring/seminorm.lean

```diff
@@ -164,6 +165,26 @@ end
 
 end ring
 
+lemma is_nonarchimedean.map_nat_cast_le_one {R F : Type*} [non_assoc_ring R]
+  [ring_seminorm_class F R] {f : F} (hf : is_nonarchimedean f) (hf1 : f 1 ≤ 1) (n : ℕ) : f n ≤ 1 :=
```

Comment:
Dumb question: is f 1 <= 1 provable from the axioms?

Reply:
Do you need `non_assoc_ring` or can you get away with a semiring? I just had a look at `ordered_semiring` and there's not much to go on; maybe hf1 is indeed needed.